### ISSUE TITLE
Missing filter error is not clear

### ISSUE THREAD

**Issue by coatless** [quarto-dev/quarto-cli#6762]
### Bug description

The description of the error message associated with a missing filter does not provide a concise statement that the filter is missing from the `_extension/` directory that should be next to the `.qmd` document.

c.f. https://github.com/coatless/quarto-webr/issues/26

### Steps to reproduce

Consider a quarto document that uses a filter:

````qmd
---
title: Simple Lightbox Example
filters:
   - lightbox
---

![A Lovely Image](mv-1.jpg){.lightbox}
````

The error message given is long and verbose without a concise statement of what is wrong.

```
Error running filter /Applications/quarto/share/filters/main.lua:
Error running filter /Users/ronin/demo-quarto/lightbox:
Could not find executable /Users/ronin/demo-quarto/lightbox
stack traceback:
	/Applications/quarto/share/filters/main.lua:4030: in function </Applications/quarto/share/filters/main.lua:4009>
	[C]: in ?
	[C]: in method 'walk'
	/Applications/quarto/share/filters/main.lua:171: in function 'run_emulated_filter'
	/Applications/quarto/share/filters/main.lua:449: in local 'callback'
	/Applications/quarto/share/filters/main.lua:454: in upvalue 'run_emulated_filter_chain'
	/Applications/quarto/share/filters/main.lua:495: in function </Applications/quarto/share/filters/main.lua:476>
stack traceback:
	/Applications/quarto/share/filters/main.lua:171: in function 'run_emulated_filter'
	/Applications/quarto/share/filters/main.lua:449: in local 'callback'
	/Applications/quarto/share/filters/main.lua:454: in upvalue 'run_emulated_filter_chain'
	/Applications/quarto/share/filters/main.lua:495: in function </Applications/quarto/share/filters/main.lua:476>
```

### Expected behavior

Emphasize that the filter desired is missing from the `_extension` folder.

**Comment by cscheid** [comment]
We've already fixed this on the 1.4 prereleases:

```
$ quarto render foo.qmd
pandoc
  to: html
  output-file: foo.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Simple Lightbox Example

FATAL (/Users/cscheid/repos/github/quarto-dev/quarto-cli/src/resources/filters/./common/wrapped-filter.lua:129) An error occurred:
Could not run /Users/cscheid/Desktop/daily-log/2023/09/07/lightbox as a JSON filter.
Please make sure the file exists and is executable.

Did you intend 'lightbox' as a Lua filter in an extension?
If so, make sure you've spelled the name of the extension correctly.

The original Pandoc error follows below.
Error running filter /Users/cscheid/Desktop/daily-log/2023/09/07/lightbox:
Could not find executable /Users/cscheid/Desktop/daily-log/2023/09/07/lightbox
```

**Comment by coatless** [comment]
Ewk! Sorry for the duplication. Great to see the improved error context.
